src/callbacks_step/time_series_dg2d.jl

```diff
@@ -68,6 +70,83 @@ function get_elements_by_coordinates!(element_ids, coordinates, mesh::TreeMesh,
     return element_ids
 end
 
+function get_elements_by_coordinates!(element_ids, coordinates,
+                                      mesh::UnstructuredMesh2D,
+                                      dg, cache)
+    if length(element_ids) != size(coordinates, 2)
+        throw(DimensionMismatch("storage length for element ids does not match the number of coordinates"))
+    end
+
+    # Reset element ids - 0 indicates "not (yet) found"
+    element_ids .= 0
+    found_elements = 0
+
+    # Iterate over all elements
+    for element in eachelement(dg, cache)
+
+        # Iterate over coordinates
+        for index in 1:length(element_ids)
+            # Skip coordinates for which an element has already been found
+            if element_ids[index] > 0
+                continue
+            end
+
+            # Construct point
+            x = SVector(ntuple(i -> coordinates[i, index], ndims(mesh)))
+
+            # Skip if point is not in the current element
+            if !is_point_in_quad(mesh, x, element)
+                continue
+            end
+
+            # Otherwise point is in the current element
+            element_ids[index] = element
+            found_elements += 1
+
+            if mesh.element_is_curved[element] && mesh.polydeg > 1
+                @warn "A time series point inside a curved element may contain errors"
+            end
+        end
+
+        # Exit loop if all elements have already been found
+        if found_elements == length(element_ids)
+            break
+        end
+    end
+
+    return element_ids
+end
+
+# For the `UnstructuredMesh2D` this uses a simple method assuming a convex
+# quadrilateral. It simply checks that the point lies on the "correct" side
+# of each of the quadrilateral's edges (basically a ray casting strategy).
+# OBS! One possibility for a more robust (and maybe faster) algorithm would
+# be to replace this function with `inpolygon` from PolygonOps.jl
+function is_point_in_quad(mesh::UnstructuredMesh2D, point, element)
+    # Helper array for the current quadrilateral element
+    corners = zeros(eltype(mesh.corners), 2, 4)
+
+    # Grab the four corners
+    for j in 1:2, i in 1:4
+        # pull the (x,y) values of these corners out of the global corners array
+        corners[j, i] = mesh.corners[j, mesh.element_node_ids[i, element]]
+    end
+
+    if cross_product(corners[:, 2] .- corners[:, 1], point .- corners[:, 1]) > 0 &&
+       cross_product(corners[:, 3] .- corners[:, 2], point .- corners[:, 2]) > 0 &&
+       cross_product(corners[:, 4] .- corners[:, 3], point .- corners[:, 3]) > 0 &&
+       cross_product(corners[:, 1] .- corners[:, 4], point .- corners[:, 4]) > 0
+        return true
+    else
+        return false
+    end
+end
+
+# 2D cross product
+function cross_product(u, v)
```

Comment:
```suggestion
@inline function cross_product_2d(u, v)
```

Reply:
Good point, this should be more explicit to avoid confusion.